internal/pkg/gateway/commit/notifier.go

```diff
@@ -58,6 +60,12 @@ func (notifier *Notifier) Notify(done <-chan struct{}, channelName string, trans
 	return notifyChannel, nil
 }
 
+// Close the notifier. This closes all notification channels obtained from this notifier. Behaviour is undefined after
+// closing and the notifier should not be used.
+func (notifier *Notifier) Close() {
+	close(notifier.cancel)
+}
```

Comment:
If Close is to be idempotent, you should consider using a sync.Once on the close.

Reply:
I don't really need to be able to close the notifier right now, and this kind of complexity is why I hadn't implemented one until previous review comments drove this "requirement". However, the use of `sync.Once` is an excellent suggestion. 👍